### ISSUE TITLE
Collect link usage data from MHV on VA.gov landing page

### ISSUE THREAD

**Issue by batemapf** [department-of-veterans-affairs/va.gov-team#54506]
## Story
"As a member of the team responsible for the MHV on VA.gov authenticated landing page, I need to know how often each of the links on the landing page is clicked, so I can better understand user flows and improve the product."

## Background
We're leaning into the Datadog RUM implementation for collecting and analyzing interaction data. While the base DD RUM implementation collects a lot of things, it appears that data about user interactions is _not_ collected when that action results in a new page being loaded, at least according to the [docs](https://docs.datadoghq.com/real_user_monitoring/browser/tracking_user_actions/#track-user-interactions). I suspect that if the page that the user navigates to is _also_ instrumented with DD RUM, then the session can be tracked and everything consolidated in Datadog for analysis. 

However, we should expect that the majority of links on the landing page will be to pages that do not have DD RUM implemented. This means we need to make sure that when a user clicks a link on the landing page (e.g., `Send a secure message`), we know about that interaction.

Potential solution here is to fire a [custom action](https://docs.datadoghq.com/real_user_monitoring/guide/send-rum-custom-actions/?tab=npm) when a link is clicked.

## Acceptance criteria
- [ ] Count of clicks on each link on the landing page can be determined in Datadog

**Comment by dcloud** [comment]
Do those other pages have any analytics or logging? Servers typically log requests which include the `Referer` header that tells us where visitors arrive from.

Using a frontend framework to track clicks that lead to new pages will impact performance for users, particularly depending on their network speeds.

In order to send custom messages (via the frontend) when a user clicks a link that would take them to a new page, you'd first have to prevent the user from visiting the new page. So you stop the browser from sending them to a new page and pause while we send data to DD via HTTP. You probably also need to wait for the HTTP call to complete; navigating to a new page might interrupt in-progress HTTP requests from the current page.

**Comment by batemapf** [comment]
all good points @dcloud, thank you. TL;DR: i think prob OK to not move forward here and just rely on data we're already getting elsewhere. im good to close it out.

some more details/answers

> Do those other pages have any analytics or logging?

they do and the data is already incorporated into VA.gov google analytics. in fact, they should be showing up in [this basic GA dashboard](https://analytics.google.com/analytics/web/#/dashboard/RIUSg1TxScOfy7EDePStHA/a50123418w177519031p176188361/)  (see `Landing page --> MHV portal pages (detail)` and `Landing page --> non-MHV portal pages (detail)
`). my over-optimizing self was trying to figure out how to get off-site link click data into datadog so we could rely exclusively/primarily on datadog for FE analytics since i find their tooling and analytical capabilities far superior to GA. but given your points, i think over-optimizing is exactly what it is.

>Using a frontend framework to track clicks that lead to new pages will impact performance for users, particularly depending on their network speeds.

not sure if there is an additional performance hit on top of the one we're already getting when pulling in the datadog RUM framework for gathering session replays and other FE analytics. either way, point taken that we need to be careful about performance.

>In order to send custom messages (via the frontend) when a user clicks a link that would take them to a new page, you'd first have to prevent the user from visiting the new page. So you stop the browser from sending them to a new page and pause while we send data to DD via HTTP. You probably also need to wait for the HTTP call to complete; navigating to a new page might interrupt in-progress HTTP requests from the current page.

my _waves hands_ hope was that the calls would not be blocking, in that if the call to datadog didn't complete in a specified time, it would be abandoned and the user would proceed to where they wanted to go. but i have no actual evidence that this is how the datadog custom action component works :-)

**Comment by dcloud** [comment]
> not sure if there is an additional performance hit on top of the one we're already getting when pulling in the datadog RUM framework for gathering session replays and other FE analytics. either way, point taken that we need to be careful about performance.

Yeah, sorry: I meant we'd be adding a performance hit via the need to pause the link action while we send an event to DD. Not that we'd be saving loading DD by not doing the custom event.

I will also say I investigated a similar request before where we tested making non-blocking/pausing custom event calls and it didn't work.

Maybe there's value in exploring adding DD RUM to MHV Classic?
